app/src/main/java/org/stepic/droid/ui/adapters/CodeToolbarAdapter.kt

```diff
@@ -10,16 +15,40 @@ import org.stepic.droid.R
 import org.stepic.droid.model.code.symbolsForLanguage
 import org.stepic.droid.ui.listeners.OnItemClickListener
 
-class CodeToolbarAdapter(private val context: Context) : RecyclerView.Adapter<CodeToolbarAdapter.CodeToolbarItem>() {
+class CodeToolbarAdapter(private val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    companion object {
+        private const val SEPARATOR_VIEW_TYPE = 1
+        private const val ELEMENT_VIEW_TYPE = 0
+    }
 
     interface OnSymbolClickListener {
         fun onSymbolClick(symbol: String)
     }
 
+    private val autocompletePrefixBackgroundSpan by lazy {
+        BackgroundColorSpan(ContextCompat.getColor(context, R.color.code_toolbar_autocomplete_prefix_background))
+    }
+
+    var autoCompleteWords: List<String> = emptyList()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
```

Comment:
using of old list size and new for updating only changed elements with `notifyItemRangeChanged` is better